File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashSet<Integer> mp = new HashSet<>() ;
        for(int i = 0 ; i< nums.length ; i++){
            if(mp.contains(nums[i]))    return true;
            mp.add(nums[i]);
        }
        return false ;
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0 , h = nums.size()-1;
        while(l <= h){
            int m = l + (h-l)/2;
            if(nums[m] == target)
                return m;
            else if(nums[m] > target)
                h--;
            else
                l++;
        }
        return -1;
    }
};
